### PR TITLE
subdomain creation and content hash setting

### PR DESCRIPTION
**Pull request by KillariDev**
image

### CODE REVIEW

**Review by MicahZoltu — APPROVED** [review]
I scanned through it and don't see anything underhanded.  May be good to have @jubalm review it a bit more thoroughly though.

File: app/ts/contenthash.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	}
+}
+
+export const tryEncodeContentHash = (contentHash: string) => {
+	try {
+		const [codec, hash] = contentHash.split('://')
+		if (codec === undefined || hash === undefined) throw new Error('invalid hash')
+		return encodeContentHash(codec, hash)
+	} catch(e) {
+		console.log(e)
```
Comment (jubalm) [inline app/ts/contenthash.ts:167]: did you want to `console.warn` or `console.error` here?

File: app/ts/contenthash.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+const isCryptographicIPNS = (cid: CID): boolean => {
+	try {
+		const { multihash } = cid
+		if (multihash.size < 38) {
+			const mh = multihashDecode(multihash.bytes)
+			if (mh.code === 0x0 && mh.size < 36) return false
+		}
+		return true
+	} catch (_) {
```
Comment (jubalm) [inline app/ts/contenthash.ts]: can be an optional catch binding here if we're not interested with the exception
```suggestion
	} catch {
```

**Review by jubalm — APPROVED** [review]

File: app/ts/contenthash.ts (hunk 1 above)
Comment (KillariDev) [reply app/ts/contenthash.ts:167]: this is removed in later prs

**Pull request merged**